README.md

```diff
@@ -1,24 +1,253 @@
 
-* Deployment instructions
+**TripTrek** 
 
-* ...
+## 🛠 Built With <a name="built-with"></a>
+
+### Tech Stack <a name="tech-stack"></a>
+
+The following stacks were used
+
+<details>
+  <summary>Technologies Used:</summary>
+  <ul>
+    <li>
+      Front-end:
+      <ul>
+        <li><a href="https://developer.mozilla.org/en-US/docs/Web/HTML">HTML</a></li>
+        <li><a href="https://developer.mozilla.org/en-US/docs/Web/CSS">CSS</a></li>
+        <li><a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript">JavaScript</a></li>
+        <li><a href="https://react-redux.js.org/">React Redux</a></li>
+        <li><a href="https://react.dev/">React</a></li>
+        <li><a href="https://webpack.js.org/">Webpack</a></li>
+      </ul>
+    </li>
+    <li>
+      Back-end:
+      <ul>
+        <li><a href="https://www.ruby-lang.org/en/">Ruby</a></li>
+        <li><a href="https://rubyonrails.org/">Rails</a></li>
+        <li><a href="https://www.postgresql.org/">PostgreSQL</a></li>
+      </ul>
+    </li>
+  </ul>
+</details>
+
+
+
+<!-- Features -->
+
+### Key Features <a name="key-features"></a>
+
+- **User Registration**
+- **Car Booking**
+- **Reserve a car**
+- **Responsive Design**
+- **Car Fleet Information**
+
+
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+
+## 🚀 Live Demo 
+<a name="live-demo" href="#">Demo(not available)</a>
+
+## 🚀 Video Demo 
+<a name="live-demo" href="#">Video(not available)</a>
+
+
+
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- GETTING STARTED -->
+
+## 💻 Getting Started <a name="getting-started"></a>
+
+### Prerequisites
+
+In order to run this project you need to have 
+- a browser and a code editor
+- `Node.js` and `npm` on your computer.
+- installed and configured ruby in your local environment:
+- setup Postgres database
+
+### Setup
+
+Clone these repositories to your desired folder:
+
+<!--
+Example commands:
+
+```sh
+  cd my-folder
+  git clone https://github.com/devalibello/final-capstone-project.git
+  git clone https://github.com/devalibello/final-capstone-project-api.git
+```
+ 
+--->
+Before to run the program, verify that you have Ruby installed on your OS running the following command:
+<!--
+```sh
+   ruby -v
+```
+--->
+
+### Usage
+
+To run the project follow these steps :
+1. Run the server :
+   1. Open your terminal and navigate to the root directory of your project
+   2. Run `bundle install`
+   3. Run `EDITOR="code --wait" rails credentials:edit`
+   4. Run `rails db:create db:migrate`
+   5. Run ```$ rails server```
+   6. open your navigator
+   7. Go to ```http://localhost:3000/```
+   8. If you receive an authentication error
+   9. Delete your `credential.yml` file
+   10. Run `rails credentials:edit`
+   11. Run ```$ rails server``` again
+2. Run the front-end :
+   1. Run the command ```$npm start ``` in your terminal.
```

Comment:
- [ ] When I setup both projects according to instructions and try to run the frontend, this happens:

I get a blank page with console errors claiming that the user is null:

![image](https://github.com/devalibello/final-capstone-project/assets/57421823/a32b017e-406d-4002-8356-8e17cf037056)

The user and the token are null in local storage:

![image](https://github.com/devalibello/final-capstone-project/assets/57421823/c7132d52-50b1-4a51-ad4a-76f53ea0271f)

This is likely because I am a new user and my local storage is not populated, therefore my user and token are null.

Here, the app will try to immediately read my token from local storage, but of course it will fail for new users:

![image](https://github.com/devalibello/final-capstone-project/assets/57421823/7c488de4-792f-4426-8693-38c8922206fe)


Kindly, fix this bug so the app does not greet the user with a blank page.

_tip:  This may work on your end because on your browser the local storage is already populated. In order to correctly debug this issue, please delete your local storage or use an incognito window._

_tip 2:  I tried to login directly by visiting localhost:3001/signup. I was able to create a new user like this._

However, upon logging in, this happens:

![image](https://github.com/devalibello/final-capstone-project/assets/57421823/02454da3-b3e3-4a13-b5c0-535e5f8338d2)

_tip 3:  The server did not report any errors, only the frontend._

_note: This means I wasn't able to run test the core functionality of the project.  This will be taken a look in the next review._